pySDC/projects/PinTSimE/switch_estimator.py

```diff
@@ -51,76 +55,66 @@ def get_new_step_size(self, controller, S):
             None
         """
 
-        self.switch_detected = False  # reset between steps
+        self.params.switch_detected = False  # reset between steps
 
         L = S.levels[0]
 
-        if not type(L.prob.params.V_ref) == int and not type(L.prob.params.V_ref) == float:
-            # if V_ref is not a scalar, but an (np.)array
-            V_ref = np.zeros(np.shape(L.prob.params.V_ref)[0], dtype=float)
-            for m in range(np.shape(L.prob.params.V_ref)[0]):
-                V_ref[m] = L.prob.params.V_ref[m]
-        else:
-            V_ref = np.array([L.prob.params.V_ref], dtype=float)
-
-        if S.status.iter > 0 and self.count_switches < np.shape(V_ref)[0]:
-            for m in range(len(L.u)):
-                if L.u[m][self.count_switches + 1] - V_ref[self.count_switches] <= 0:
-                    self.switch_detected = True
-                    m_guess = m - 1
-                    break
+        if S.status.iter == S.params.maxiter:
 
-            if self.switch_detected:
-                t_interp = [L.time + L.dt * self.coll_nodes_local[m] for m in range(len(self.coll_nodes_local))]
+            self.params.switch_detected, m_guess, vC_switch = L.prob.get_switching_info(L.u, L.time)
 
-                vC_switch = []
-                for m in range(1, len(L.u)):
-                    vC_switch.append(L.u[m][self.count_switches + 1] - V_ref[self.count_switches])
+            if self.params.switch_detected:
+                t_interp = [L.time + L.dt * self.params.coll_nodes[m] for m in range(len(self.params.coll_nodes))]
 
                 # only find root if vc_switch[0], vC_switch[-1] have opposite signs (intermediate value theorem)
                 if vC_switch[0] * vC_switch[-1] < 0:
 
-                    self.t_switch = self.get_switch(t_interp, vC_switch, m_guess)
+                    self.params.t_switch = self.get_switch(t_interp, vC_switch, m_guess)
 
                     # if the switch is not find, we need to do ... ?
-                    if L.time < self.t_switch < L.time + L.dt:
-                        r = 1
-                        tol = self.dt_initial / r
+                    if L.time < self.params.t_switch < L.time + L.dt:
 
-                        if not np.isclose(self.t_switch - L.time, L.dt, atol=tol):
-                            dt_search = self.t_switch - L.time
+                        if not np.isclose(self.params.t_switch - L.time, L.dt, atol=self.params.tol):
+                            dt_switch = self.params.t_switch - L.time
+                            self.log(
+                                f"Located Switch at time {self.params.t_switch:.6f} is outside the range of tol={self.params.tol:.4f}",
```

Comment:
I am guessing the tolerance is sth like 1e-3. For such numbers you can use `e` instead of `f` in string formatting to get the exponent representation.